projects/plugins/jetpack/extensions/blocks/subscriber-login/subscriber-login.php

```diff
@@ -72,16 +72,16 @@ function get_current_url() {
 function get_subscriber_login_url( $redirect ) {
 	$redirect = ! empty( $redirect ) ? $redirect : get_site_url();
 
-	// Copied from projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php
+	// Taken from projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php and simplified a bit
```

Comment:
Why not try use from shared location right away?

Reply:
The next step could be to move it into a shared location. Right now we have it as a private function in `Abstract_Token_Subscription_Service`:
https://github.com/Automattic/jetpack/blob/b64a145116591e19018cb143a2154f1ac748d3c1/projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-abstract-token-subscription-service.php#L521-L525
But the logic for building log in URL for a simple site is different.

Reply:
For simple sites we need to use the `wpcom_logmein_redirect_url` function, at least this is what the Paywall block is doing.